src/sas/sascalc/fit/fitstate.py

```diff
@@ -0,0 +1,408 @@
+    disambiguate different SASentry sections with the same dataset.
+    """
+    p_model, s_model = state.formfactorcombobox, state.structurecombobox
+    if s_model is not None and s_model != "" and s_model.lower() != "none":
+        return '*'.join((p_model, s_model))
+    else:
+        return p_model
+
+def get_data_weight(state):
+    """
+    Get error bars on data.  These could be the values computed by reduction
+    and stored in the file, the square root of the intensity (if instensity
+    is approximately counts), the intensity itself (would be better as a
+    percentage of the intensity, such as 2% or 5% depending on relative
+    counting time), or one for equal weight uncertainty depending on the
+    value of state.dI_*.
+    """
+    # Cribbed from perspectives/fitting/utils.py:get_weight and
+    # perspectives/fitting/fitpage.py: get_weight_flag
+    weight = None
+    if state.enable2D:
+        dy_data = state.data.err_data
+        data = state.data.data
+    else:
+        dy_data = state.data.dy
+        data = state.data.y
+    if state.dI_noweight:
+        weight = np.ones_like(data)
+    elif state.dI_didata:
+        weight = dy_data
+    elif state.dI_sqridata:
+        weight = np.sqrt(np.abs(data))
+    elif state.dI_idata:
+        weight = np.abs(data)
+    return weight
+
+_MODEL_CACHE = {}
+def load_model(name):
+    """
+    Given a model name load the Sasview shim model from sasmodels.
+
+    If name starts with "[Plug-in]" then load it as a custom model from the
+    plugins directory.  This code does not go through the Sasview model manager
+    interface since that loads all available models rather than just those
+    needed.
+    """
+    # Remember the models that are loaded so they are only loaded once.  While
+    # not strictly necessary (the models will use identical but different model
+    # info structure) it saves a little time and memory for the usual case
+    # where models are reused for simultaneous and batch fitting.
+    if name in _MODEL_CACHE:
+        return _MODEL_CACHE[name]
+    if name.startswith(PLUGIN_NAME_BASE):
+        name = name[len(PLUGIN_NAME_BASE):]
+        plugins_dir = find_plugins_dir()
+        path = os.path.abspath(os.path.join(plugins_dir, name + ".py"))
+        #print("loading custom", path)
+        model = load_custom_model(path)
+    elif name and name is not None and name.lower() != "none":
+        #print("loading standard", name)
+        model = _make_standard_model(name)
+    else:
+        model = None
+    _MODEL_CACHE[name] = model
+    return model
+
+def parse_optional_float(value):
+    """
+    Convert optional floating point from string to value, returning None
+    if string is None, empty or contains the word "None" (case insensitive).
+    """
+    if value is not None and value != "" and value.lower() != "none":
+        return float(value)
+    else:
+        return None
```

Comment:
or just try... except on float(value) catching ValueErrors. Otherwise you risk not just two return values (float or None) but also unchecked exception if 'value' is weird.

Reply:
I wouldn't want to silently ignore badly formatted float values, which is what would happen with `try: return float(value) except: return None`.   We could do finer grained checking as we are loading in each field, but in this case I'm happy to reject the entire file if the any data in it is bad.

Reply:
What do you mean by "silently ignore?"  I would think that would in fact be the right answer?  assuming it replaces with a NAN so it could be caught later if so desired?  I have found users get rather miffed if an otherwise perfectly good file barfs because of one bad value that could have been ignored and allowed them to continue working.  But as I said ... I probably am missing the point entirely here?

Reply:
The code in question is used for parsing the value, min, and max fields for each parameter in a saved fit.  Since these are values that we've written to the save file, they ought to be in a format that is parseable.

One case that should not be parsed as a float is enumerated parameters, which store the value as a string such as `Rexp(-|nu|z)`.  If we are not handling this properly, I want it to raise an error so that we know rather than silently treating it as None which would then get interpreted as the default first choice.  If we treat it as NaN this would then raise an error when we try to convert it to an int, which is far away from where the actual problem occurred.    If we are going to fail, then fail early.  

If we are going to accept the input then we better be sure it is correct, otherwise it will require an observant user to notice that the value has changed.  Particularly in this case, where the saved file may be processed on a remote fitting server and the user never sees the value.